src/libcollections/btree/map.rs

```diff
@@ -875,20 +874,30 @@ impl<K: Ord, V: Ord> Ord for BTreeMap<K, V> {
     }
 }
 
-#[stable]
-impl<K: Show, V: Show> Show for BTreeMap<K, V> {
-    fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
-        try!(write!(f, "BTreeMap {{"));
-
-        for (i, (k, v)) in self.iter().enumerate() {
-            if i != 0 { try!(write!(f, ", ")); }
-            try!(write!(f, "{:?}: {:?}", *k, *v));
-        }
+macro_rules! fmt_btree_map {
+    ($($Trait:ident),*) => {
+        $(
+            impl<K: fmt::$Trait, V: fmt::$Trait> fmt::$Trait for BTreeMap<K, V> {
+                fn fmt(&self, f: &mut fmt::Formatter) -> fmt::Result {
+                    try!(write!(f, "BTreeMap {{"));
+
+                    for (i, (k, v)) in self.iter().enumerate() {
+                        if i != 0 { try!(write!(f, ", ")); }
+                        try!(fmt::$Trait::fmt(k, f));
```

Comment:
Nah basically the only thing we'd want to require is that the same formatting trait was used (there's no great way to transitively specify flags for sub-components). So for example an implementation of `LowerHex` would use `format_args!("{:x}", arg)` for intermediate elements.